### ISSUE TITLE
Renaming the async methods with Async suffixes

### ISSUE THREAD

**Issue by Thieum** [mycroes/Sally7#3]
Considering the convention documented [in the MSDN](https://docs.microsoft.com/en-us/dotnet/csharp/async#important-info-and-advice) , it would feel more natural to have the async methods defined in the library renamed with an Async suffix:

```
        public Task Open();
        public Task Read(params IDataItem[] dataItems);
        public Task Write(params IDataItem[] dataItems);
        private async Task<int> ReadTpkt() /* would be not breaking */
```

to 

```
        public Task OpenAsync();
        public Task ReadAsync(params IDataItem[] dataItems);
        public Task WriteAsync(params IDataItem[] dataItems);
        private async Task<int> ReadTpktAsync() /* would be not breaking */
```

It's a breaking change, so before submitting a PR, I wanted to know if it was something you had interest in, and the best course of action if that was the case.

since the library is still pre 1.0.0, breaking changes would be acceptable regarding semver, but there is always the possibility to keep the old signature for 1 version with an obsolete attribute too.

**Comment by mycroes** [comment]
This was one of the things I still wanted to do as well. Feel free to create a PR for this. There's no need to keep the old methods around, it's pre-1.0 and I don't think there are many users out there. At work I'm using a wrapper, so for me it's only a single change as well.

Regards,
Michael

**Comment by mycroes** [comment]
Fixed by b44c912e6193e81795368466f91220a9f58d395d.

**Comment by Thieum** [comment]
@mycroes thanks a lot!
